src/views/Hearing/HearingContainer.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {Switch, Route, Redirect} from 'react-router-dom';
+import Section from '../../components/Hearing/Section/SectionContainer';
+import Header from '../../components/Hearing/Header';
+import WrappedCarousel from '../../components/Carousel';
+import {getHearingWithSlug} from '../../selectors/hearing';
+import PropTypes from 'prop-types';
+import {fetchHearing as fetchHearingAction, setLanguage as setLanguageAction} from '../../actions';
+import LoadSpinner from '../../components/LoadSpinner';
+import isEmpty from 'lodash/isEmpty';
+import { injectIntl, intlShape } from 'react-intl';
+import {SectionTypes} from '../../utils/section';
+import {canEdit} from '../../utils/hearing';
+import HearingEditor from '../../components/admin/HearingEditor';
+import * as HearingEditorSelector from '../../selectors/hearingEditor';
+import { fetchHearingEditorMetaData } from '../../actions/hearingEditor';
+import {getUser} from '../../selectors/user';
+import config from '../../config';
+
+export class HearingContainerComponent extends React.Component {
+  componentWillMount() {
+    const {fetchHearing, fetchEditorMetaData, match: {params}} = this.props;
+    fetchHearing(params.hearingSlug);
+    fetchEditorMetaData();
+  }
+
+  render() {
+    const {
+      hearing,
+      intl,
+      language,
+      match,
+      user,
+      labels,
+      hearingDraft,
+      hearingLanguages,
+      isLoading,
+      contactPersons,
+      setLanguage
+    } = this.props;
+    const mainSectionId = !isEmpty(hearing) ? hearing.sections.find(section => section.type === SectionTypes.MAIN).id : null;
+    const reportUrl = config.apiBaseUrl + '/v1/hearing/' + hearing.slug + '/report';
+
+    return (
+      <div className="hearing-page">
+
+        {!isEmpty(hearing) ?
+          <div>
+            {(!isEmpty(user) && canEdit(user, hearing)) &&
+              <HearingEditor
+                hearing={hearingDraft}
+                hearingLanguages={hearingLanguages}
+                labels={labels}
+                user={user}
+                isLoading={isLoading}
+                contactPersons={contactPersons}
+              />
+            }
+            <div className="hearing-wrapper" id="hearing-wrapper">
+              <Header
+                hearing={hearing}
+                activeLanguage={language}
+                intl={intl}
+                reportUrl={reportUrl}
+                setLanguage={setLanguage}
+              />
+              <WrappedCarousel hearing={hearing} intl={intl} language={language}/>
+              <Switch>
+                <Route path="/:hearingSlug/:sectionId" component={Section} />
+                <Redirect from="/:hearingSlug" to={`/${match.params.hearingSlug}/${mainSectionId}`} />
```

Comment:
Here, we have the problem that when entering the hearing, the user is redirected to the section url.

Basically the hearing main page is used by city employees to copy the hearing URL everywhere to external sources. Therefore, the hearing page URL must be 1) short, 2) memorable and 3) typable.

I'm afraid we will have to have a separate route for the main section that will render the main section in the absence of section id.